src/data_store/generic_data_store.cpp

```diff
@@ -76,6 +76,18 @@ generic_data_store::generic_data_store(generic_data_reader *reader, model *m) :
   if (opts->has_bool("verbose") && opts->get_bool("verbose")) {
     m_verbose = true;
   }
+
+  if (opts->has_string("use_tarball")) {
+    m_dir = m_reader->get_local_file_dir();
+  }
+
+  if (m_comm->get_num_models() != 1) {
```

Comment:
I'm not sure I understand your question, but I'll try to answer. I assume all models use the same data set (that may be a bad assumption). The problem is that there are many places where the data_store code may break if there is more than one model. Since all (or almost all) of our model_zoo models use a single model (procs_per_model: 0 in the prototext file), for an initial implementation I assumed there is a single model (hope that makes sense).

Reply:
We have multi-model cases (e.g., LTFB) where data store is of importance. If this is an initial implementation may be it is ok to make assumption of a single model.